models/repo_generate.go

```diff
@@ -144,7 +264,7 @@ func GenerateGitHooks(ctx DBContext, templateRepo, generateRepo *Repository) err
 			return err
 		}
 
-		generateHook.Content = templateHook.Content
+		generateHook.Content = generateExpansion(templateHook.Content, templateRepo, generateRepo)
```

Comment:
So hooks are always expanded?

----
#### _"Just thinking to myself" mode start_

Maybe use `.giteatemplate` like:

```
.git/hooks/pre-receive
.git/hooks/**
```

That could never clash with actual repository content because it's illegal.

This would only be allowed for globs that start with `.git/hooks/` that 

#### _"Just thinking to myself" mode end_
----

Anyway, I'm worried that some hooks may not be suitable for expansion (e.g. hooks that were not written by the template creator but inherited from someone else?)

Reply:
This would come down to the template owner and whether you "trusted" them enough to use their hooks imo  
As with the other comments, things not meant for expansion should be escaped. I can add some documentation to help template owners, though.

Alternatively, variable expansion itself could be another option when generating a repository.

Reply:
But hooks are _invisible_ unless you have permissions to edit them.

Reply:
I think if this is really a concern we should never expand hooks (for now).  
If the template owner cannot view/edit them, they also have no way of reliably adding lines to a `.giteatemplate` to say otherwise.

Reply:
@jolheiser But the hooks may be there because some admin _needs_ the users to have those hooks.... I imagine a future where users without special permissions can't create their own repos unless they use an org approved template. 😉

Anyway, I agree that this is a sensitive issue. Maybe hooks are not a good choice for a first implementation of variable expansion.